roles/matrix-synapse/tasks/update_user_password.yml

```diff
@@ -0,0 +1,48 @@
+---
+
+- name: Fail if playbook called incorrectly
+  fail:
+    msg: "The `username` variable needs to be provided to this playbook, via --extra-vars"
+  when: "username is not defined or username == '<your-username>'"
+
+- name: Fail if playbook called incorrectly
+  fail:
+    msg: "The `password` variable needs to be provided to this playbook, via --extra-vars"
+  when: "password is not defined or password == '<your-password>'"
+
+- name: Fail if not using matrix-postgres container
+  fail:
+    msg: "This command is working only when matrix-postgres container is being used"
+  when: "not matrix_postgres_enabled"
+
+- name: Ensure matrix-synapse is started
+  service:
+    name: matrix-synapse
+    state: started
+    daemon_reload: yes
+  register: start_result
+
+- name: Ensure matrix-postgres is started
+  service:
+    name: matrix-postgres
+    state: started
+    daemon_reload: yes
+  register: postgres_start_result
+
+
+- name: Wait a while, so that Matrix Synapse can manage to start
+  pause:
+    seconds: 7
+  when: start_result.changed
+
+- name: Wait a while, so that Matrix Postgres can manage to start
+  pause:
+    seconds: 7
+  when: postgres_start_result.changed
+
+- name: Generate password hash
+  shell: "/usr/local/bin/matrix-synapse-generate-password-hash {{ password }}"
```

Comment:
Wouldn't it be possible to replace this custom script with `/usr/bin/docker exec -it matrix-synapse /usr/local/bin/hash_password -c /data/homeserver.yaml -p {{ password }}`

It's very cool how it uses `expect` to help us doing it non-interactively, but if we can, it'd be nicer to avoid it (less code & no need to install `expect`).

-------

We may even be able to have just one single script (called `matrix-postgres-set-user-password`) which does everything all at once (generates the hash and runs the SQL). I can see how keeping it separate may have some value too, but it may also be causing confusion (too many scripts in `/usr/local/bin/`).

Reply:
I have missed the existence of the `-p` option... I'll change it and will implement it with one script. There is no point of having a `matrix-synapse-generate-password-hash` when there is a docker command for that.

Reply:
Great!

Reply:
The command now uses the '-p' option to generate the password hash. `matrix-synapse-generate-password-hash` script and `expect` requirements have been removed.